### ISSUE TITLE
Make dataloader more flexible

### ISSUE THREAD

**Issue by sfmig** [SainsburyWellcomeCentre/crabs-exploration#108]
Probably a good place to start the refactoring. 

- [ ] We would like to easily filter out videos or frames that we don't want in the training / test set.
- [ ] We would also like to easily define different shuffles for the test / train sets.

Right now we are ensuring the dataloader is the same in train and eval by setting the same seed - ideally there is a way around this (one option may be to run train and eval simultaneously)
- [ ] Investigate options ☝ 
- [ ] Implement anything that looks reasonable.

**Comment by sfmig** [comment]
Since we are lightning, [the datamodule](https://lightning.ai/docs/pytorch/stable/data/datamodule.html) seems v useful.
